File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/base/BaseFragment.java
package hr.fer.oobl.iorder.iorder.base;

import hr.fer.oobl.iorder.iorder.injection.fragment.DaggerFragment;

public abstract class BaseFragment extends DaggerFragment implements BaseView {

    @Override
    public void onStop() {
        getPresenter().stop();
        super.onStop();
    }

    @Override
    public void onPause() {
        getPresenter().deactivate();
        super.onPause();
    }

    public abstract ScopedPresenter getPresenter();
}

File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/base/BaseFragment.kt
package hr.fer.oobl.iorder.iorder.base


import hr.fer.oobl.iorder.iorder.injection.fragment.DaggerFragment

abstract class BaseFragment : DaggerFragment(), BaseView {

    abstract val presenter: ScopedPresenter

    override fun onStop() {
        presenter.stop()
        super.onStop()
    }

    override fun onPause() {
        presenter.deactivate()
        super.onPause()
    }
}
